app/renderer/components/Config/Config.js

```diff
@@ -0,0 +1,59 @@
+import { ipcRenderer, remote } from 'electron';
+import React, { Component } from 'react';
+import { Input, Row, Col, Button } from 'antd';
+import styles from './Config.css';
+
+const {app, dialog, getCurrentWindow} = remote;
+
+export default class Config extends Component {
+
+  componentWillMount () {
+    this.props.getEnvironmentVariables();
+  }
+
+  saveAndRestart () {
+    const {environmentVariables} = this.props;
+    ipcRenderer.send('appium-save-env', environmentVariables);
+    ipcRenderer.once('appium-save-env-done', () => {
+      const message = `Application must be restarted for changes to take effect`;
+      const dialogOptions = {type: 'info', buttons: ['Restart Now', 'Restart Later'], message};
+      dialog.showMessageBox(dialogOptions, (response) => {
+        if (response === 0) {
+          // If 'Restart Now' restart the application
+          app.relaunch();
+          app.exit();
```

Comment:
why do we both relaunch and exit?

Reply:
Not sure why we have to do that but apparently it's required: https://github.com/electron/electron/issues/6579

Reply:
ok, as long as it's the way we're supposed to do it!